sverweij/dependency-cruiser
# Add updated deprecation list of core modules

This PR should fix #373 

## Description

After checking Node.js list of deprecated core modules for version 10, 12, 14, I updated it to the latest version.

## Motivation and Context

It allows to notify user about planned deprecations for core-module usages in node up to 14 vertsion

## How Has This Been Tested?

1. `yarn install dependency-cruiser -g`
2. run `dependency-cruiser --init`
3. update list `path: "^(punycode|domain|constants|sys|_linklist|_stream_wrap)$",` in created config to new multiline path.
4.  add script `src/script.js`

```javascript
// you can change to any other module from list
const splayTree = require('v8/tools/splaytree');

console.log(splayTree);
```
5. install/switch node to versions 8/10/12/14
6. run script `node ./script.js`
7. run dep-cruiser checks via `dependency-cruiser --config .dependency-cruiser.js src`
8. check the results

## Screenshots

<img width="799" alt="dep-cruiser" src="https://user-images.githubusercontent.com/13730032/96387455-edaae300-1199-11eb-9faa-4061af80b97f.png">


## Types of changes



- [ ] Refactor (non-breaking change which fixes an issue without changing functionality)

## Checklist




- [x] :book:

  - [x] My change doesn't require a documentation update, or ...
  - it _does_ and I have updated it

- [x] :balance_scale:
  - The contribution will be subject to [The MIT license](../LICENSE), and I'm OK with that.
  - The contribution is my own original work.
  - I am ok with the stuff in [**CONTRIBUTING.md**](./CONTRIBUTING.md).
